statchatexpress/app.js

```diff
@@ -0,0 +1,75 @@
+import express from 'express';
+import cors from 'cors';
+import morgan from 'morgan';
+import { sequelize } from './database.js';
+import { User, Stats } from './models/index.js';
+
+const app = express();
+
+app.use(cors())
+app.use(express.json()); // Middleware for parsing JSON bodies from HTTP requests
+app.use(morgan('combined'))
+
+// Route to get all users
+app.get('/users', async (req, res) => {
+  try {
+    const users = await User.findAll();
+    res.json(users);
+  } catch (err) {
+    res.status(500).json({ message: err.message });
+  }
+});
+
+// Route to get a user by id
+app.get('/users/:id', async (req, res) => {
+  try {
+    const user = await User.findByPk(req.params.id);
+    if (user) {
+      res.json(user);
+    } else {
+      res.status(404).json({ message: 'User not found' });
+    }
+  } catch (err) {
+    res.status(500).json({ message: err.message });
+  }
+});
+
+// Route to get all stats, with associated users
+app.get('/stats', async (req, res) => {
+  try {
+    const stats = await Stats.findAll({
+      include: [{ model: User, as: 'user' }],
+      order: [['createdAt', 'DESC']]
+    });
+    res.json(stats);
+  } catch (err) {
+    res.status(500).json({ message: err.message });
+  }
+});
+
+// Route to create a new post
+app.post('/stats', async (req, res) => {
+  try {
+    const stats = await Stats.create(req.body);
+
+    const statsWithUser = await Stats.findOne({
+      where: { id: stats.id },
+      include: [{ model: User, as: 'user' }]
+    });
+
+    res.status(201).json(statsWithUser);
+  } catch (err) {
+    res.status(500).json({ message: err.message });
+  }
+});
+
+sequelize.sync({ alter: true })
```

Comment:
I'm not familiar with sequelize - would you mind explaining for me what this does? :) What does the `alter` parameter do?

Reply:
this makes it so that the database and the code match, if there are any differences in the code then the database will update. lets say i add a new property to the user model in the code: age. it will automatically change the property in the database

Reply:
ah gotcha. That's convenient!